pkgs/development/libraries/epoxy/default.nix

```diff
@@ -1,41 +1,60 @@
-{ lib, stdenv, fetchFromGitHub, autoreconfHook, pkg-config, utilmacros, python3
-, libGL, libX11
+{ lib
+, stdenv
+, fetchFromGitHub
+, meson
+, ninja
+, pkg-config
+, utilmacros
+, python3
+, libGL
+, libX11
 }:
 
-with lib;
+let
+  inherit (lib) getLib optional optionalString;
 
+in
```

Comment:
Wouldn't it be easier to just use lib.optional?

Reply:
Just like we define our inputs to our calls to `mkDerivation` instead of passing in `pkgs`, being specific about what we use from `lib` is a Nice-Thing(tm).

Reply:
I don't think that is comparable. pkgs is mainly not passed in because overwriting a single package in it is a quite expensive operation. Because we never overwrite anything in lib we can pass it in as is.

Reply:
That doesn't change the fact that being specific about what we use makes it easier to read and understand. In any case, as we don't have a formal style guide, I'm doing it this way as I prefer it and there are no objective reasons not to.